synapse/util/manhole.py

```diff
@@ -81,7 +81,10 @@ def manhole(username, password, globals):
     checker = checkers.InMemoryUsernamePasswordDatabaseDontUse(**{username: password})
 
     rlm = manhole_ssh.TerminalRealm()
-    rlm.chainedProtocolFactory = lambda: insults.ServerProtocol(
+    # mypy ignored here because:
+    # - can't deduce types of lambdas
+    # - variable is Type[ServerProtocol], expr is Callable[[], ServerProtocol]
+    rlm.chainedProtocolFactory = lambda: insults.ServerProtocol(  # type: ignore[misc,assignment]
```

Comment:
It's probably better to create a real temporary class here:

```python
class SynapseServerProtocol(insults.ServerProtocol):
    def __init__(self):
        super().__init__(SynapseManhole, dict(globals, __name__="__console__"))
```

I think should work?

Reply:
I tried something of that sort, it then said (paraphrased): found `Type[SynapseServerProtocol]` expected `Type[ServerProtocol]`.